pkg/controller/datasource-controller.go

```diff
@@ -102,8 +103,26 @@ func (r *DataSourceReconciler) update(ctx context.Context, dataSource *cdiv1.Dat
 		if isReady {
 			updateDataSourceCondition(dataSource, cdiv1.DataSourceReady, corev1.ConditionTrue, "DataSource is ready to be consumed", ready)
 		}
+	} else if sourceSnapshot := dataSource.Spec.Source.Snapshot; sourceSnapshot != nil {
+		snapshot := &snapshotv1.VolumeSnapshot{}
+		ns := cc.GetNamespace(sourceSnapshot.Namespace, dataSource.Namespace)
+		isReady := false
+		if err := r.client.Get(ctx, types.NamespacedName{Namespace: ns, Name: sourceSnapshot.Name}, snapshot); err != nil {
+			if !k8serrors.IsNotFound(err) {
+				return err
+			}
+			r.log.Info("Snapshot not found", "name", sourceSnapshot.Name)
+			updateDataSourceCondition(dataSource, cdiv1.DataSourceReady, corev1.ConditionFalse, "Snapshot not found", cc.NotFound)
+		} else if snapshot.Status != nil && snapshot.Status.ReadyToUse != nil && *snapshot.Status.ReadyToUse {
+			isReady = true
```

Comment:
why not get rid of `isReady` and call `updateDataSourceCondition` here?

Reply:
true, done